syntax = "proto3";

package user;

message User {
    string id = 1;
    string name = 2;
    string email = 3;
    string password = 4;
}

message GetUserRequest {
    string id = 1;
}

message CreateUserRequest {
    User user = 1;
}

message UpdateUserRequest {
    User user = 1;
}

message DeleteUserRequest {
    string id = 1;
}

service UserService {
    rpc GetUser(GetUserRequest) returns (User);
    rpc CreateUser(CreateUserRequest) returns (User);
    rpc UpdateUser(UpdateUserRequest) returns (User);
    rpc DeleteUser(DeleteUserRequest) returns (User);
}
